cmake/modules/unittest.cmake

```
@@ -4,12 +4,23 @@ cmake_minimum_required(VERSION 3.20.0)
 
 enable_language(C CXX ASM)
 
+# Custom targets for compiler and linker flags.
+# These are normally created under FindTargetTools, but we can't include it here because unit
+# testing doesn't support cross compiling.
+add_custom_target(asm)
+add_custom_target(compiler)
+add_custom_target(compiler-cpp)
+add_custom_target(linker)
+
+include(FindHostTools)
 include(root)
 include(boards)
 include(arch)
 include(configuration_files)
 include(kconfig)
 
+include(${ZEPHYR_BASE}/cmake/target_toolchain_flags.cmake)
```

then update to target tools usage here:
```suggestion
find_package(TargetTools)

enable_language(C CXX ASM)
 
include(${ZEPHYR_BASE}/cmake/target_toolchain_flags.cmake)
```

and in order to have target tools working, you need to update this line to ensure cross-compilation doesn't kick in:
https://github.com/zephyrproject-rtos/zephyr/blob/96f1ee93fc7b1d7785fd34b098077fa19fd57bbb/cmake/modules/FindHostTools.cmake#L69
to this:
```
if("${ARCH}" STREQUAL "posix" OR "${ARCH}" STREQUAL "unit_testing")
```